service/history/tasks/sync_workflow_state_task.go

```diff
@@ -38,7 +38,8 @@ type (
 		definition.WorkflowKey
 		VisibilityTimestamp time.Time
 		TaskID              int64
-		Version             int64
+		// TODO: validate this version in source task converter
+		Version int64
```

Comment:
Context: This task is for data backfill of immutable wf so we don't check this version.